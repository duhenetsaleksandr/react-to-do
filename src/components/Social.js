import React from 'react';
import { ReactComponent as TelegramLogo } from '../images/telegram.svg';
import { ReactComponent as InstagramLogo } from '../images/instagram.svg';

const getTelegramLogo = () => ( <TelegramLogo className="social-link" /> );
const getInstagramLogo = () => ( <InstagramLogo className="social-link" /> );

class SocialLink extends React.Component {
    render() {
        return (
            <a href={this.props.link} target="_blank" rel="noopener noreferrer" title={this.props.title}>
                {this.props?.child()}
            </a>
        );
    }
}

export default class Social extends React.Component {
    render() {
        return (
            <div className="social-icons">
                <SocialLink link="https://t.me/duhenets" title="telegram" child={getTelegramLogo}/>
                <SocialLink link="https://www.instagram.com/duhenets_aleksandr/" title="instagram" child={getInstagramLogo}/>
            </div>
        );
    }
}
