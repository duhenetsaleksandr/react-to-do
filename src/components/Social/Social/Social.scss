@use 'src/styles/colors';

.social-icons {
    display: flex;
}

.social-icons >:not(:last-child) {
    margin-right: 10px;
}

.social-link {
    width: 25px;
    height: 25px;
    transition: all 0.2s ease;

    &.dark {
        color: colors.$secondary-color;
    }

    &.light {
        color: colors.$primary-color;
    }

    &:hover {
        cursor: pointer;
        color: colors.$helper-color;
    }
}
