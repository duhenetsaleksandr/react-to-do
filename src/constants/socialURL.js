export const telegramURL = 'https://t.me/duhenets';
export const instagramURL = 'https://www.instagram.com/duhenets_aleksandr/';
