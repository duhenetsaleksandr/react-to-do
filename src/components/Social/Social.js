import React from 'react';
import './social.scss';
import SocialLink from './SocialLink';
import { ReactComponent as TelegramLogo } from '../../images/telegram.svg';
import { ReactComponent as InstagramLogo } from '../../images/instagram.svg';
import * as socialURL from '../../constants/socialURL';

const Social = () => (
    <div className="social-icons">
        <SocialLink link={socialURL.telegramURL} title="telegram">
            <TelegramLogo className="social-link" />
        </SocialLink>
        <SocialLink link={socialURL.instagramURL} title="instagram">
            <InstagramLogo className="social-link" />
        </SocialLink>
    </div>
);

export default Social;
