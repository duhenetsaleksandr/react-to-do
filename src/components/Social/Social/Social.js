import React, { useContext } from 'react';
import ThemeContext from 'Context/ThemeContext';
import './Social.scss';
import SocialLink from 'components/Social/SocialLink';
import { ReactComponent as TelegramLogo } from 'images/telegram.svg';
import { ReactComponent as InstagramLogo } from 'images/instagram.svg';
import * as socialURL from 'constants/socialURL';

const Social = () => {
    const { darkTheme } = useContext(ThemeContext);

    return (
        <div className="social-icons">
            <SocialLink link={socialURL.telegramURL} title="telegram">
                <TelegramLogo className={`social-link ${darkTheme ? 'dark' : 'light'}`} />
            </SocialLink>
            <SocialLink link={socialURL.instagramURL} title="instagram">
                <InstagramLogo className={`social-link ${darkTheme ? 'dark' : 'light'}`} />
            </SocialLink>
        </div>
    );
};

export default Social;
