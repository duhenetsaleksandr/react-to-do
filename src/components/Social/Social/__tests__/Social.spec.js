import React from 'react';
import Social from '../Social';
import { instagramURL, telegramURL } from "constants/socialURL";

describe('should render Social component', () => {
    let component;
    beforeEach(() => {
        component = mount(<Social/>);
    });

    it('should contain wrapper with .social-icons class', () => {
        const wrapper = component.find('div.social-icons');
        expect(wrapper).toHaveLength(1);
    });

    it('should contain two SocialLink component', () => {
        const socialLink = component.find('SocialLink');
        expect(socialLink).toHaveLength(2);
    });

    it('should first SocialLink component contain telegram', () => {
        const socialLink = component.find('SocialLink').at(0);
        expect(socialLink.getElement().props.link).toBe(telegramURL);
        expect(socialLink.getElement().props.title).toBe('telegram');
    });

    it('should second SocialLink component contain instagram', () => {
        const socialLink = component.find('SocialLink').at(1);
        expect(socialLink.getElement().props.link).toBe(instagramURL);
        expect(socialLink.getElement().props.title).toBe('instagram');
    });

    it('should render Social component snapshot', () => {
        const component = shallow(<Social/>);
        expect(component).toMatchSnapshot();
    });
});
