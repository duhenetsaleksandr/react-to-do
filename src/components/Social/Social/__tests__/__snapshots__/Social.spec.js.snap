// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Social component should render Social component snapshot 1`] = `
<div
  className="social-icons"
>
  <SocialLink
    link="https://t.me/duhenets"
    title="telegram"
  >
    <ForwardRef(SvgTelegram)
      className="social-link dark"
    />
  </SocialLink>
  <SocialLink
    link="https://www.instagram.com/duhenets_aleksandr/"
    title="instagram"
  >
    <ForwardRef(SvgInstagram)
      className="social-link dark"
    />
  </SocialLink>
</div>
`;
